### PR TITLE
chore(clippy): Simplify minor stuff found by clippy

### PR DESCRIPTION
**Pull request by klausi**
Executed clippy 0.0.191, fixed the trivial stuff.
Filed https://github.com/rust-lang-nursery/rust-clippy/issues/2615 for all the format_err!() false positives.

### CODE REVIEW

File: src/cargo/ops/cargo_new.rs [hunk 1]
```diff
@@ -444,7 +444,7 @@ fn mk(config: &Config, opts: &MkOptions) -> CargoResult<()> {
 
     match vcs {
         VersionControl::Git => {
-            if !fs::metadata(&path.join(".git")).is_ok() {
+            if fs::metadata(&path.join(".git")).is_err() {
```
Comment (matklad) [inline src/cargo/ops/cargo_new.rs]: !.is_ok reads somewhat better to me...

Maybe let’s make it radically better and extract a 

    let exists = |dotdir: &str| -> bool

function while we are at it?

Comment (matklad) [reply src/cargo/ops/cargo_new.rs]: @alexcrichton looks like we use to ways to check if a file exists in cargo:

* `Path.exists()`
* `fs::metadata()::is_ok()`

Which one is preferred and why? :)

Comment (alexcrichton) [reply src/cargo/ops/cargo_new.rs]: Oh I'd prefer `exists` today. Usage of `metadata` predates the existence of `exists`

**Comment by klausi** [comment]
@mikerite suggested in https://github.com/rust-lang-nursery/rust-clippy/issues/2615 that we could use failure::err_msg() instead of format_err!() when there are no variable replacements. I'm a bit on the fence if that is worth it.

Pro: using format_err!() when there is nothing to format is confusing and failure::err_msg() should be sued instead.
Con: spreading both format_err!() and failure::err_msg() for raising errors around the code base is confusing, there should be only way for consistency.

What do you think?

**Comment by matklad** [comment]
I'd use `format_err!` for uiformity, and to reduce the diff when you add/remove variables.

**Comment by klausi** [comment]
OK, then I leave all format_err!() untouched. Fixed a couple more simply thing in crates-io, this should be ready now!

**Pull request merged**